test/acceptance/features/steps/openshift.py

```diff
@@ -334,3 +335,25 @@ def search_resource_lst_in_namespace(self, resource_plural, name_pattern, namesp
             return lst
         print('Resource list is empty under namespace - {}'.format(namespace))
         return None
+
+    def get_secret_data(self, secret_name, secret_key, namespace, secret_value):
+        json_path = f'.data.{secret_key}'
+        if secret_value == "true":
+            start = 0
```

Comment:
agree with @pratikjagrut - in particular we should not do pooling only if some value is true.

Reply:
the polling in the jsonPath is to fetch the specific key from the secret data. In this case, as I had mentioned earlier, the secret key is present always but with a value nil. It takes time for the CustomReady:<no value> to turn to CstomReady:true. If I just depend on the polling of the secret key, then I will always receive the key instantaneously as it is already present in the secret. What actually I am waiting upon here is for the ,no value. to change to true, which does not happen instantaneously

Reply:
ok, if that is the case, we should first wait until secret itself gets update and then to read the value. We read the secret two times during this test, so we know what was the previous generation.

Reply:
I'm noticing that somehow this polling is not working as expected it is working just as sleep(). When you get the first true value it should break the while loop but instead, it is running till timeout.

According to the logic max number of iteration of while loop should be = (timeout/interval) + 1(value of start is 0) = (180/5)+1=37

Here I ran your code and printed the output if count those lines it will be equal to 37.
```
>>>>>>>>> Output while >>>>>>>>>>> PG5vIHZhbHVlPg==
>>>>>>>>> Output while >>>>>>>>>>> PG5vIHZhbHVlPg==
>>>>>>>>> Output while >>>>>>>>>>> PG5vIHZhbHVlPg==
>>>>>>>>> Output while >>>>>>>>>>> PG5vIHZhbHVlPg==
>>>>>>>>> Output while >>>>>>>>>>> PG5vIHZhbHVlPg==
>>>>>>>>> Output while >>>>>>>>>>> PG5vIHZhbHVlPg==
>>>>>>>>> Output while >>>>>>>>>>> PG5vIHZhbHVlPg==
>>>>>>>>> Output while >>>>>>>>>>> PG5vIHZhbHVlPg==
>>>>>>>>> Output while >>>>>>>>>>> PG5vIHZhbHVlPg==
>>>>>>>>> Output while >>>>>>>>>>> PG5vIHZhbHVlPg==
>>>>>>>>> Output while >>>>>>>>>>> PG5vIHZhbHVlPg==
>>>>>>>>> Output while >>>>>>>>>>> PG5vIHZhbHVlPg==
>>>>>>>>> Output while >>>>>>>>>>> PG5vIHZhbHVlPg==
>>>>>>>>> Output while >>>>>>>>>>> PG5vIHZhbHVlPg==
>>>>>>>>> Output while >>>>>>>>>>> PG5vIHZhbHVlPg==
>>>>>>>>> Output while >>>>>>>>>>> PG5vIHZhbHVlPg==
>>>>>>>>> Output while >>>>>>>>>>> PG5vIHZhbHVlPg==
>>>>>>>>> Output while >>>>>>>>>>> PG5vIHZhbHVlPg==
>>>>>>>>> Output while >>>>>>>>>>> PG5vIHZhbHVlPg==
>>>>>>>>> Output while >>>>>>>>>>> PG5vIHZhbHVlPg==
>>>>>>>>> Output while >>>>>>>>>>> PG5vIHZhbHVlPg==
>>>>>>>>> Output while >>>>>>>>>>> PG5vIHZhbHVlPg==
>>>>>>>>> Output while >>>>>>>>>>> PG5vIHZhbHVlPg==
>>>>>>>>> Output while >>>>>>>>>>> PG5vIHZhbHVlPg==
>>>>>>>>> Output while >>>>>>>>>>> PG5vIHZhbHVlPg==
>>>>>>>>> Output while >>>>>>>>>>> PG5vIHZhbHVlPg==
>>>>>>>>> Output while >>>>>>>>>>> PG5vIHZhbHVlPg==
>>>>>>>>> Output while >>>>>>>>>>> PG5vIHZhbHVlPg==
>>>>>>>>> Output while >>>>>>>>>>> PG5vIHZhbHVlPg==
>>>>>>>>> Output while >>>>>>>>>>> dHJ1ZQ==
>>>>>>>>> Output while >>>>>>>>>>> dHJ1ZQ==
>>>>>>>>> Output while >>>>>>>>>>> dHJ1ZQ==
>>>>>>>>> Output while >>>>>>>>>>> dHJ1ZQ==
>>>>>>>>> Output while >>>>>>>>>>> dHJ1ZQ==
>>>>>>>>> Output while >>>>>>>>>>> dHJ1ZQ==
>>>>>>>>> Output while >>>>>>>>>>> dHJ1ZQ==
>>>>>>>>> Output while >>>>>>>>>>> dHJ1ZQ==
>>>>>>>>> Output Last >>>>>>>>>>> dHJ1ZQ==
```
I believe this `dHJ1ZQ==` is base64 encoded value for `true`. 
So while should've terminated at first occurrence only.

So @Avni-Sharma  if I'm not wrong here about my findings we might need to use a different mechanism for polling.
Possibly as @pedjak mentioned above using generations we can try.

[Here are the complete log and changes I made to `get_secret_data` of test run.](https://gist.github.com/pratikjagrut/32599a6aa5d23c5477b0f3efca07d5c1)

Reply:
@pedjak it seems secret has no generatio field in metadata
```
apiVersion: v1
data:
  BACKEND_HOST: ZXhhbXBsZS5jb21tb24=
  BACKEND_READY: dHJ1ZQ==
  CustomHost: ZXhhbXBsZS5jb21tb24=
  CustomReady: dHJ1ZQ==
kind: Secret
metadata:
  creationTimestamp: "2020-09-03T14:25:48Z"
  managedFields:
  - apiVersion: v1
    fieldsType: FieldsV1
    fieldsV1:
      f:data:
        .: {}
        f:BACKEND_HOST: {}
        f:BACKEND_READY: {}
        f:CustomHost: {}
        f:CustomReady: {}
      f:metadata:
        f:ownerReferences:
          .: {}
          k:{"uid":"c4232d50-9be4-43be-9589-617e28b555fc"}:
            .: {}
            f:apiVersion: {}
            f:controller: {}
            f:kind: {}
            f:name: {}
            f:uid: {}
      f:type: {}
    manager: service-binding-operator-local
    operation: Update
    time: "2020-09-03T14:26:58Z"
  name: binding-request-backend
  namespace: test-namespace-11
  ownerReferences:
  - apiVersion: apps.openshift.io/v1alpha1
    controller: true
    kind: ServiceBindingRequest
    name: binding-request-backend
    uid: c4232d50-9be4-43be-9589-617e28b555fc
  resourceVersion: "357650"
  selfLink: /api/v1/namespaces/test-namespace-11/secrets/binding-request-backend
  uid: 8bcfc157-6362-4e52-887a-82b63265acbb
type: Opaque
```

Reply:
:thinking:

Reply:
@Avni-Sharma you can use `resourceVersion` instead: https://kubernetes.io/docs/reference/using-api/api-concepts/#efficient-detection-of-changes

Reply:
afaik Reading the generation is also oc GET call and for retrieving data of a secret also, it is oc GET. Whether we fetch resourceVersion or the secret data it will have the same impact on efficiency and will still work. In that case, unless the efficiency is altered I would prefer not to make that change.

Reply:
Updated timeout https://github.com/redhat-developer/service-binding-operator/pull/597/commits/f0136cc8565b4aa248f5dafc58922680e88edd03

Reply:
Done